src/unix/unix_c/unix_get_pw_gr_nam_id_job.c

```diff
@@ -81,9 +82,17 @@ static value alloc_group_entry(struct group *entry)
     {                                                                  \
         size_t buffer_size = sysconf(_SC_##CONF##_R_SIZE_MAX);         \
         if (buffer_size == (size_t)-1) buffer_size = 16384;            \
-        job->buffer = (char *)lwt_unix_malloc(buffer_size);            \
-        job->result = FUNC##_r(job->ARG, &job->entry, job->buffer,     \
-                               buffer_size, &job->ptr);                \
+        while (1)                                                   \
+        {                                                              \
+            job->buffer = (char *)lwt_unix_malloc(buffer_size);        \
+            job->result = FUNC##_r(job->ARG, &job->entry, job->buffer, \
+                                   buffer_size, &job->ptr);            \
+            if (job->result == 0) break;                               \
+            if (job->result != ERANGE) break;                          \
+            free(job->buffer);                                         \
```

Comment:
I think we risk a double `free` here. If the result is `ERANGE` but the buffer size reaches the limit, we will have freed the buffer and not allocated a new one. `result_##FUNC` will then attempt to free the buffer again. We should probably only `free` the buffer in this worker function if the next buffer is still not too large, i.e. move this line to be last in the loop body.

Reply:
I guess I forgot what C memory management is like when I said this loop is simple :P

Reply:
Fixed. Good catch!

Yeah, I haven't done C in a long time and I'd forgotten the hassle… I'm leaving this one unresolved because of that.